freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/flow/FreeIpaPostInstallService.java

```diff
@@ -132,14 +135,30 @@ private void setInitialFreeIpaPolicies(Stack stack, FreeIpaClient freeIpaClient)
         }
         freeIpaClient.addPermissionToPrivilege(USER_ADMIN_PRIVILEGE, SET_PASSWORD_EXPIRATION_PERMISSION);
         freeIpaPermissionService.setPermissions(stack, freeIpaClient);
-        if (!Objects.equals(MAX_USERNAME_LENGTH, freeIpaClient.getUsernameLength())) {
-            LOGGER.debug("Set maximum username length to {}", MAX_USERNAME_LENGTH);
-            freeIpaClient.setUsernameLength(MAX_USERNAME_LENGTH);
-        }
+        Config ipaConfig = freeIpaClient.getConfig();
+        updateMaxUsernameLength(freeIpaClient, ipaConfig);
+        updateMaxHostnameLength(stack, freeIpaClient, ipaConfig);
         passwordPolicyService.updatePasswordPolicy(freeIpaClient);
         modifyAdminPasswordExpirationIfNeeded(freeIpaClient);
     }
 
+    private static void updateMaxUsernameLength(FreeIpaClient freeIpaClient, Config ipaConfig) throws FreeIpaClientException {
+        if (!Objects.equals(MAX_NAME_LENGTH, ipaConfig.getIpamaxusernamelength())) {
+            LOGGER.debug("Set maximum username length to {}", MAX_NAME_LENGTH);
+            freeIpaClient.setUsernameLength(MAX_NAME_LENGTH);
+        }
+    }
+
+    private static void updateMaxHostnameLength(Stack stack, FreeIpaClient freeIpaClient, Config ipaConfig) throws FreeIpaClientException {
+        Integer maxHostNameLength = stack.getPrimaryGateway().map(im -> im.getDomain().length() + MAX_DOMAIN_PART_LENGTH).orElse(MAX_NAME_LENGTH);
```

Comment:
I think you should have a Math.min(MAX_NAME_LENGTH, calculatedhostnamelength) so it wouldn't try to set an invalid value if domain is too long

Reply:
I don't understand this, because `stack.getPrimaryGateway()` returns with an optional so this way I handled the case when it doesn't present then falling back to 255 chars not to have the ~63 chars default value on IPA side.

Reply:
thats fine, but you have to limit it. eg right now if you have a domain which is 200 character long, you would add this extra 63/64 and it would be rejected by FreeIPA as the max value is 255